build/build/src/engine/context.rs

```diff
@@ -523,24 +523,24 @@ impl RunContext {
         }
 
 
-        // if build_native_runner {
```

Comment:
This code has been introduced by 
- #3843

Looking at the commits in the PR it looks like @mwu-tow tried to add the test, but failed. Can we enable such test now? Do you know how to [fix the failure](https://github.com/enso-org/enso/actions/runs/7354059528/job/20020788572?pr=8641#step:8:29), @mwu-tow?

Can we also expand the test to run twice and once use `ENSO_JAVA=espresso`, @mwu-tow?